### PR TITLE
added rum support. fixed compile errors in ratom. (fixes issue #27)

### PR DESCRIPTION
**Pull request by bamarco**

### CODE REVIEW

File: src/posh/lib/ratom.cljc [hunk 1]
```diff
@@ -17,146 +17,147 @@
...
+   (defmacro add! [x v]
+     (if-cljs &env `(.push ~x ~v)
+                   `(.add ~(with-meta x {:tag 'java.util.ArrayList}) ~v))))
 
 #?(:clj
-(defmacro array-list [& args]
-  (if-cljs &env `(array ~@args)
-                `(doto (ArrayList.) ~@(for [arg args] `(.add ~arg))))))
+   (defmacro array-list [& args]
+     (if-cljs &env `(into [] ~@args)
```
Comment (alexandergunnarson) [inline src/posh/lib/ratom.cljc]: This is going to cause issues with ‘alength’  and ‘aget’ further down, as well as a few other things. A JS array is extensible so it’s basically an arraylist. Persistent vectors’ properties are too dissimilar to just swap it in here.

Comment (bamarco) [reply src/posh/lib/ratom.cljc]: The array macro doesn't seem to exist in cljs anymore. What is the correct function to use then?

**Comment by mpdairy** [comment]
Nice, is this ready to merge then? I don't have clojure set up on this computer and didn't write any tests anyways so I can't test it.

**Comment by ojw** [comment]
Hey what's the status of this PR?  Is there anything an outsider can do to help?  Rum support in Posh would be great.

**Comment by metasoarous** [comment]
@ojw I'll defer to @bamarco and @alexandergunnarson on this question, but to my recollection things are more or less ready to go and just need to be tested out. If you're comfortable with Rum, it might be super helpful to have you test things out in coordination with Brian and Alexander.

File: src/posh/lib/ratom.cljc (hunk 1 above)
Comment (alexandergunnarson) [reply src/posh/lib/ratom.cljc]: Sorry, don't know why I didn't respond to this! Just wanted to say that `array` is definitely still there — see also https://github.com/clojure/clojurescript/blob/master/src/main/clojure/cljs/core.cljc#L2482.

Comment (alexandergunnarson) [reply src/posh/lib/ratom.cljc]: Though I think I might know why it's not compiling properly — try:
```
`(cljs.core/array ~@args)
```

rather than
```
`(array ~@args)
```

Because of how CLJS-in-CLJ compilation works, possibly the `array-list` macro was resolving `array` to `clojure.core/array` rather than `cljs.core/array`. Not sure — I'd have to test it out.

File: src/posh/plugin_base.cljc [hunk 2]
```diff
@@ -99,19 +99,20 @@
...
-                                        (fn [_ _]
-                                          ;;(println "no DISPOSING: " storage-key)
-                                          (when-not (= (:cache options) :forever)
-                                            (swap! posh-atom
-                                                   (fn [posh-atom-val]
-                                                     (assoc (p/remove-item posh-atom-val storage-key)
-                                                       :ratoms (dissoc (:ratoms posh-atom-val) storage-key)
-                                                       :reactions (dissoc (:reactions posh-atom-val) storage-key)))))))]
+                 query-reaction       ((:derive-reaction dcfg) [query-ratom] storage-key (fn [q] q))]
+                ;  query-reaction       ((:make-reaction dcfg)
```
Comment (alexandergunnarson) [inline src/posh/plugin_base.cljc]: I haven't thought through what would happen if you change the implementation of `query-reaction` to what you have now, but assuming your version works, could you @bamarco remove the commented portion of code?

File: src/posh/rum.cljc [hunk 3]
```diff
@@ -0,0 +1,31 @@
+(ns posh.rum
+  (:require 
+    [posh.plugin-base :as base]
+    [rum.core :as rum]
+    [datascript.core :as d]))
+
+(defn make-reaction [f & {:as local-mixin :keys [on-dispose]}]
+  ; (prn "make-reaction" (str f))
```
Comment (alexandergunnarson) [inline src/posh/rum.cljc]: Looks like this is a lingering todo @bamarco?

File: src/posh/rum.cljc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  ;; TODO: handle at least the on-dispose method of the local mixin
+  (rum/derived-atom [] ::no-refs-anyways f))
+
+; (defn make-reaction2 [f & {:as local-mixin :keys [on-dispose]}]
```
Comment (alexandergunnarson) [inline src/posh/rum.cljc]: Is this commented code block okay to be removed @bamarco?

**Comment by alexandergunnarson** [comment]
@ojw @metasoarous @bamarco I haven't tested out any part of this PR yet but in theory yes, it should be ready to go once we figure out some of the concerns I posted in the review comments. @ojw I think it would be a big help to add some tests! Since the repo appears to have none, then even one good one would help :) Thanks!

**Comment by bamarco** [comment]
Moving the pr to the new repo

**Comment by bamarco** [comment]
@ojw finally finished the rum plugin. it is pending in the new posh repository
